69.md

```diff
@@ -0,0 +1,293 @@
+
+Example NIP-57 zap request:
+
+```json
+{
+  "id": "<event_id>",
+  "pubkey": "<sender_pubkey>",
+  "created_at": 1234567890,
+  "kind": 21001,
+  "tags": [
+    ["p", "<receiver_pubkey>"]
+  ],
+  "content": "<NIP-44 encrypted {\"offer\":\"<zap1234>\",\"zap\":<kind 9734 zap request event>}>",
+  "sig": "<signature>"
+}
+```
+
+There is no inherent requirement on the service to acknowledge zap requests, since spontaneous payments are default behavior. However, both Clients and Service implementations SHOULD consider offer identifiers, where `startsWith("zap")`, indication that the service will honor zap requests for that offer. A service should not error if no zap request was included by the sender on a zap indicated offer, the service should treat it as any other spontaneous payment.
+
+## General Process Flow
+
+1. **Payer Scans or Clicks the Static Payment Code**
+2. **Payer's Wallet Decodes the Payment Code**
+3. **Payer's Wallet Sends a Nostr Event to the Specified Relay**
+   - Addressed to the receiver's public key, containing the offer identifying string.
+   - Event Type: Ephemeral Kind 21001 | NIP-44 Encrypted
+   - **Conditional**: Include the amount in sats the sender wishes to pay if the payment type is spontaneous.
+   - Optional: Include additional payer data.
+4. **Receiver Responds with Lightning Invoice**
+   - Generates a Lightning invoice and responds with a Nostr event containing the invoice details.
+   - Event Type: Ephemeral Kind 21001 | NIP-44 Encrypted
+   - **Content**: `{"bolt11":"<BOLT11_invoice_string>"}`
+   - Optional: Include additional purchase data.
+5. **Payer Pays the Invoice**
+   - Completes the payment by settling the Lightning invoice.
+6. **Optional: Receiver Emits Payment Receipt**
+   - This step is out of scope for this NIP, but potential receipt considerations include:
+     - Nostr Event as Receipt: Emitting a public Nostr event as a "zap" receipt for verifiable record-keeping.
+     - Lightning Pre-Image: The Lightning pre-image is proof of payment.
+     - External / WebHook: The receiver may finalize the flow out-of-band.
+
+### Nostr Events
+
+This NIP specifies the use of event kind `21001` with the following structure:
+
+- `content`: NIP-44 encrypted payment details.
+- `tags`: 
+  - `p`: Receiver's public key (hex).
+  - `e`: Used in response to the requesting event.
+
+Example request event:
+
+```json
+{
+  "id": "<event_id>",
+  "pubkey": "<sender_pubkey>",
+  "created_at": 1234567890,
+  "kind": 21001,
+  "tags": [
+    ["p", "<receiver_pubkey>"]
+  ],
+  "content": "<NIP-44 encrypted offer identifier and conditional payer data>",
+  "sig": "<signature>"
+}
+```
+
+The `content` field, after NIP-44 decryption, should contain an object with the following structure:
+
+```json
+{
+    "offer": "<offer_string>",
+    "amount": "<conditional_amount_in_sats>",
+    "payer_data": "<optional_payer_data>"
+}
+```
+
+Example response event with `bolt11` invoice:
```

Comment:
Check out this [LUD-22 payee data](https://github.com/lnurl/luds/pull/252) proposal. It may be useful for the receiver to be able to dynamically communicate free-form data back to the sender if they request certain fields, etc.

Reply:
Yea as mentioned above I leaned into keeping the scope wallets only, but also leaving room for upgrades... particularly in the context of nostr as opposed to lnurl, nostr itself is a better place to tie those requirements to a product/service addressable event rather than link encoding (will probably break QR's etc)